File: PartnerApi/Ruby/b2_create_group_member.rb
# frozen_string_literal: true

require 'net/http'
require 'uri'
require 'json'

api_url = '' # Provided by b2_authorize_account
auth_token = '' # Provided by b2_authorize_account
admin_account_id = '' # Provided by b2_authorize_account
group_id = '' # Provided by b2_list_groups
member_email = 'new-group-member@mycompany.com' # Email address for account being created

uri = URI("#{api_url}/b2api/v4/b2_create_group_member")

request = Net::HTTP::Post.new(uri)
request['Authorization'] = auth_token
request['Content-Type']  = 'application/json'
request.body = { adminAccountId: admin_account_id, groupId: group_id, memberEmail: member_email }.to_json

response = Net::HTTP.start(uri.host, uri.port, use_ssl: uri.scheme == 'https') do |http|
  http.request(request)
end

puts response.body

File: PartnerApi/Java/B2CreateGroupMember.java
package com.backblaze.examples;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class B2CreateGroupMember {
    public static void main(String[] args) {
        String apiUrl = ""; // Provided by b2_authorize_account
        String accountAuthorizationToken = ""; // Provided by b2_authorize_account
        String adminAccountId = ""; // Provided by b2_authorize_account
        String groupId = ""; // Provided by b2_list_groups
        String memberEmail = "new-group-member@mycompany.com"; // Email address for account being created

        String postParams = "{\"adminAccountId\":\"" + adminAccountId + "\",\"groupId\":\"" + groupId + "\",\"memberEmail\":\"" + memberEmail + "\"}";
        try (HttpClient client = HttpClient.newHttpClient()) {
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(apiUrl + "/b2api/v4/b2_create_group_member"))
                    .header("Authorization", accountAuthorizationToken)
                    .header("Content-Type", "application/json")
                    .POST(HttpRequest.BodyPublishers.ofString(postParams))
                    .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            String jsonResponse = response.body();
            System.out.println(jsonResponse);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
